Views/Widget-Hero.liquid

```diff
@@ -1,12 +1,22 @@
+{% assign backgroundType = Model.ContentItem.Content.Hero.BackgroundType.Text %}
+
 <section class="section section--hero section--center">
     <div class="section__content">
         <div class="hero">
             <div class="hero__body">
                 <h1 class="hero__title">{{ Model.ContentItem.Content.Hero.DisplayText.Text }}</h1>
                 <a href="{{ Model.ContentItem.Content.Hero.CTAUrl.Text | href }}" class="btn__btn btn btn--primary">{{ Model.ContentItem.Content.Hero.CTALabel.Text }}</a>
             </div>
-
-            {{ Model.Content | shape_render }}
+            
+            {% if backgroundType == "embed" %}
+                <div class="hero__media">
+                    <div class="embed">
+                        <iframe src="{{ Model.ContentItem.Content.Hero.BackgroundEmbed.Text }}" frameborder="0" webkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>
```

Comment:
How comes you don't think it will work out the box? I use a Vimeo embed in the example content that's created from the set up recipe.

https://github.com/EtchUK/Etch.OrchardCore.ThemeBoilerplate/blob/master/content/Recipes/setup.recipe.json#L579

Reply:
That's cool then, I know different platforms work differently with how they like embeds. I was thinking that some providers had dropped or were dropping support for iframes, but with your example and I just checked on youtube I think it's fine.

One thing to note from a users standpoint is you can't just drop in a YT video link and expect it to work, you'll have to specifically get the embed URL, e.g.:
https://www.youtube.com/embed/grkOmLDj8gI instead of https://www.youtube.com/watch?v=grkOmLDj8gI

Reply:
Because it's using a `TextField` we can't really do much to help the user with entering the YouTube embed link. In the future we could potentially look to create a new content field that is specific for specifying third party video embed urls. That content field could validate the common providers which would definitely improve the UX for users who aren't familiar with specifying the embed URL.

Reply:
Aye, I think its fine for now and in future, we can enhance the UX for this 😄

Reply:
For reference, Umbraco comes with several pre-made providers ( https://github.com/umbraco/Umbraco-CMS/tree/v8/dev/src/Umbraco.Web/Media/EmbedProviders,), usually using OEmbed where available to generate embed markup

Reply:
@PhilDye Very useful!